gdx/src/com/badlogic/gdx/graphics/g3d/utils/CameraHelper.java

```diff
@@ -0,0 +1,216 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.g3d.utils;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Mesh;
+import com.badlogic.gdx.graphics.VertexAttribute;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.g3d.Material;
+import com.badlogic.gdx.graphics.g3d.Renderable;
+import com.badlogic.gdx.graphics.g3d.RenderableProvider;
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+import com.badlogic.gdx.utils.Pool;
+
+/** Camera helper helps you to visualize your camera. Of course, you shouldn't use camera helper with your rendering camera.
+ * <p>
+ * How to use it:
+ * </p>
+ * <pre>
+ * // Create helper
+ * CameraHelper helper = new CameraHelper(camera);
+ * // During the rendering
+ * camera.update();
+ * helper.update();
+ * modelBatch.render(helper);
+ * // After using it
+ * helper.dispose();
+ * </pre>
+ * @author realitix */
+public class CameraHelper implements RenderableProvider, Disposable {
+	private Camera camera;
+	private Mesh mesh;
+	private Color frustumColor = new Color(1, 0.66f, 0, 1);
+	private Color coneColor = new Color(1, 0, 0, 1);
+	private Color upColor = new Color(0, 0.66f, 1, 1);
+	private Color targetColor = new Color(1, 1, 1, 1);
+	private Color crossColor = new Color(0.2f, 0.2f, 0.2f, 1);
+	private float vertices[];
+	private int id;
+	private Vector3 tmp = new Vector3();
+	private Vector3 tmp2 = new Vector3();
+	private Renderable renderable = new Renderable();
+
+	/** Create camera helper
+	 * @param camera */
+	public CameraHelper (Camera camera) {
+		this.camera = camera;
+		init();
+	}
+
+	private void init () {
+		// Init indices
+		short[] indices = new short[] {
+			// Near
+			0, 1, 1, 2, 2, 3, 3, 0,
+			// Far
+			4, 5, 5, 6, 6, 7, 7, 4,
+			// Sides
+			0, 4, 1, 5, 2, 6, 3, 7,
+			// Cone
+			0, 8, 1, 8, 2, 8, 3, 8,
+			// Cross near
+			9, 10, 11, 12,
+			// Cross far
+			13, 14, 15, 16,
+			// Target (position -> near -> far)
+			8, 17, 17, 18,
+			// Up triangle
+			19, 2, 2, 3, 3, 19};
+
+		// Init mesh
+		int maxVertices = 20;
+		mesh = new Mesh(false, maxVertices, indices.length, new VertexAttribute(Usage.Position, 3, "a_position"),
+			new VertexAttribute(Usage.ColorPacked, 4, "a_color"));
+		mesh.setIndices(indices, 0, indices.length);
+
+		// Init vertices
+		vertices = new float[maxVertices * (mesh.getVertexSize() / 4)];
+
+		// Init renderable
+		renderable.meshPart.mesh = mesh;
+		renderable.meshPart.offset = 0;
+		renderable.meshPart.size = mesh.getNumIndices();
+		renderable.meshPart.primitiveType = GL20.GL_LINES;
+		renderable.material = new Material();
```

Comment:
Probably a good idea to `update` it after construction so it can never be in an undefined state.